app/models/user.rb

```diff
@@ -3,4 +3,18 @@ class User < ApplicationRecord
   has_many :ranked_works, through: :votes, source: :work
 
   validates :username, uniqueness: true, presence: true
+  # validates :uid, presence: true
+  # validates :provider, presence: true
+
+  def self.build_from_github(auth_hash)
+    user = User.new
+    user.uid = auth_hash[:uid]
+    user.provider = 'github'
+    user.username = auth_hash[:info][:name]
```

Comment:
Just so you know, it's perfectly legal on Github to have an account with no `:name` field set.